### PR TITLE
gh-110745: pathlib.Path.read_text add a newline argument

### PR DESCRIPTION
**Pull request by Okabe-Junya**
```
gh-110745: pathlib.Path.read_text add a newline argument
```

* Issue: gh-110745

----
:books: Documentation preview :books:: https://cpython-previews--110880.org.readthedocs.build/

### CODE REVIEW

**Comment by barneygale** [comment]
Looking good! Could you also add a test in `Lib/test/test_pathlib.py`? Maybe have a look at `test_write_text_with_newlines` :)

**Comment by Okabe-Junya** [comment]
Thank you for your kind review! I apologize for the delayed response.

As you suggested, I intend to add the appropriate tests. Please give me a moment.

**Comment by Okabe-Junya** [comment]
Only the test on Windows is failing. I wonder why...

**Comment by Okabe-Junya** [comment]
Implementing tests using os.linesep seems challenging. Especially on Windows, where the newline code is `\r\n`, covering this in test cases appears difficult.

Would it be appropriate to separate test cases for each OS?

**Comment by barneygale** [comment]
Regarding `newline=None`, this should convert all newlines to `\n` on read, on all OSs, so your test shouldn't need to reference `os.linesep`. Maybe something like this?

```python
    def test_read_text_with_newlines(self):
        p = self.cls(BASE, 'fileA')
        p.write_bytes(b'abcde\r\nfghlk\n\rmnopq')
        self.assertEqual(p.read_text(newline=None),   'abcde\nfghlk\n\nmnopq')
        self.assertEqual(p.read_text(newline=''),     'abcde\r\nfghlk\n\rmnopq')
        self.assertEqual(p.read_text(newline='\r'),   'abcde\r\nfghlk\n\rmnopq')
        self.assertEqual(p.read_text(newline='\n'),   'abcde\r\nfghlk\n\rmnopq')
        self.assertEqual(p.read_text(newline='\r\n'), 'abcde\r\nfghlk\n\rmnopq')
```

**Comment by Okabe-Junya** [comment]
@barneygale 

Thank you for your thoughtful comment!!

I've implemented tests based on your feedback, and they now pass!!

**Comment by rawwar** [comment]
closes #110745

File: Doc/library/pathlib.rst [hunk 1]
```diff
@@ -1329,6 +1329,8 @@ call fails (for example because the path doesn't exist).
 
    .. versionadded:: 3.5
 
+   .. versionchanged:: 3.12
```
Comment (barneygale) [inline Doc/library/pathlib.rst]:
```suggestion
   .. versionchanged:: 3.13
```

File: Misc/NEWS.d/next/Library/2023-10-15-08-08-26.gh-issue-110745.mxEkh0.rst [hunk 2]
```diff
@@ -0,0 +1 @@
+Added *newline* parameter to ``pathlib.Path.read_text()``.
```
Comment (barneygale) [inline Misc/NEWS.d/next/Library/2023-10-15-08-08-26.gh-issue-110745.mxEkh0.rst]:
```suggestion
Added *newline* parameter to :meth:`pathlib.Path.read_text`.
Patch by Junya Okabe.
```

**Review by barneygale — CHANGES_REQUESTED** [review]
Sorry for my delayed feedback. Last thing to do is to fix the version number, and give yourself some credit in the changelog if you wish :)

**Comment by Okabe-Junya** [comment]
Thank you for your kind review!! (I might not have mentioned properly. I am sorry for that)

I accept your polite suggestion and am ready to merge this. Thank you.

**Comment by Okabe-Junya** [comment]
I have made the requested changes; please review again

**Review by barneygale — APPROVED** [review]
Looks good, thank you very much!

**Pull request merged**